### PR TITLE
MR-24 Regression plot page

### PR DESCRIPTION
**Pull request by JanionNevill**
Added regression plot page.

### CODE REVIEW

File: R/meta_regression/regression_plot_panel.R [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+      actionButton(inputId = ns("remove_comparator_btn"), label = "Remove"),
+      div(
+        style = "float: right;",
+        actionButton(inputId = ns("remove_all_btn"), label = "Remove all")
+      ),
+      
+      h3("Plot Options"),
+      # Covariate value
+      .AddTooltip(
+        checkboxInput(inputId = ns("covariate"), label = "Show covariate value", value = TRUE),
```
Comment (nabury) [inline R/meta_regression/regression_plot_panel.R:54]: I really like the tooltips but it isn't obvious that they are there. Is there any way of making it more obvious without having a messy looking ui?

File: R/meta_regression/regression_plot_panel.R [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+      .AddTooltip(
+        checkboxInput(inputId = ns("contributions"), label = "Show contributions", value = TRUE),
+        tooltip = "Show study contributions as circles, where a bigger circle represents a larger contribution"
+      ),
+      
+      div(
+        id = ns("contribution_options"),
+        p("Study circle sized by:"),
+        div(
+          .CreateInlineBlock("% Contribution", style = "padding-right: 10pt;"),
```
Comment (nabury) [inline R/meta_regression/regression_plot_panel.R:89]: The labelling for this "jumps around" on smaller screen widths
Screenshot 2023-12-21 at 14 21 34

File: R/meta_regression/regression_plot_panel.R [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+      ),
+      # Contributions
+      .AddTooltip(
+        checkboxInput(inputId = ns("contributions"), label = "Show contributions", value = TRUE),
+        tooltip = "Show study contributions as circles, where a bigger circle represents a larger contribution"
+      ),
+      
+      div(
+        id = ns("contribution_options"),
+        p("Study circle sized by:"),
```
Comment (nabury) [inline R/meta_regression/regression_plot_panel.R:87]: This doesn't get "ghosted out" like the slider which makes it seem like it's in use all the time

File: R/meta_regression/regression_plot_panel.R [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+          .CreateInlineBlock("% Contribution", style = "padding-right: 10pt;"),
+          shinyWidgets::materialSwitch(
+            inputId = ns("contribution_toggle"),
+            inline = TRUE,
+            status = "primary"
+          ),
+          .CreateInlineBlock("Inverse Variance")
+        ),
+        div(
+          .CreateInlineBlock(
```
Comment (nabury) [inline R/meta_regression/regression_plot_panel.R:98]: I like the disabling of the widgets but here, where there are two linked to the "show contributions" tick box it is less obvious that the circle size multiplier is linked

**Review by nabury — CHANGES_REQUESTED** [review]
The 'study circle sized by' widget needs fixing to display properly on narrower screens as it's confusing (a shinyWidgets radio group button could be an option so that the text is displayed on the switch rather than beside it).

I don't find the linking of the two widgets that are disabled by the 'show contributions' selector really intuitive/obvious but I suspect it's one of those things that everyone would have an opinion on and it would be difficult to get a perfect solution.

File: R/meta_regression/regression_plot_panel.R (hunk 2 above)
Comment (JanionNevill) [reply R/meta_regression/regression_plot_panel.R]: I've changed these to both be standard inputs, so they behave better.

File: R/meta_regression/regression_plot_panel.R (hunk 3 above)
Comment (JanionNevill) [reply R/meta_regression/regression_plot_panel.R:87]: That's a limitation of the widget. I've changed it to be a standard input, so it behaves now.

File: R/meta_regression/regression_plot_panel.R (hunk 4 above)
Comment (JanionNevill) [reply R/meta_regression/regression_plot_panel.R]: It should be more obvious now that all the disabling is fixed.

File: R/meta_regression/covariate_value_panel.R [hunk 5]
```diff
@@ -14,43 +14,50 @@
...
-      ns = ns,
-      .CreateInlineBlock(
-        numericInput(
-          inputId = ns("numeric"),
-          label = NULL,
-          min = -.Machine$double.xmax,
-          max = .Machine$double.xmax,
-          value = 0,
-          width = "100pt"
+    title = "This value will be used by all regression analysis output",
```
[the remark this answers is not in the document]
Comment (JanionNevill) [reply R/meta_regression/covariate_value_panel.R:17]: This is intended to be here as it applies to all of the tabs.

File: R/covariate_analysis_panel.R [hunk 6]
```diff
@@ -36,7 +36,7 @@ covariate_analysis_panel_ui <- function(id) {
                 style = "color: orange;"
```
[the remark this answers is not in the document]
Comment (JanionNevill) [reply R/covariate_analysis_panel.R:36]: The click menus are good for when there is a lot of information,. or formatted text. When the information is minimal and all on one line, a standard tooltip is generally preferred.

File: R/meta_regression/regression_plot_panel.R (hunk 1 above)
Comment (JanionNevill) [reply R/meta_regression/regression_plot_panel.R]: I've added tooltip icons to make it a little more clear.

**Review by nabury — APPROVED** [review]
I like the changes. Fixed a typo I spotted.

**Pull request merged**